horovod/run/run.py

```diff
@@ -779,47 +793,49 @@ def _run(args):
         if settings.verbose >= 2:
             print('SSH was successful into all the remote hosts.')
 
-    if len(remote_host_names) > 0:
-        if settings.verbose >= 2:
-            print('Testing interfaces on all the hosts.')
+    common_intfs = None
+    # Skipping interface discovery for LSF cluster as it slows down considerably the job start
+    if not lsf.LSFUtils.using_lsf():
```

Comment:
This block is getting pretty heavily nested.  Maybe we can pull this out into a utility function `_get_common_interfaces`, where the first couple lines are a precondition check:

```
if lsf.LSFUtils.using_lsf():
    return None
```

What do you think?

Reply:
Yes I agree, code will be cleaner. I will do that.